### ISSUE TITLE
AGPLv3?

### ISSUE THREAD

**Issue by sean-** [ahri/flask-mustache#1]
Is there any particular reason that this is AGPLv3 instead of MIT like the rest of Flask?

**Comment by ahri** [comment]
Sorry, I completely missed this due to some gmail labs thing misclassifying my email as "bulk". Bye bye Smart Labels!

Rather than talk about why I chose AGPL3, I'll explain why I like the GPL over MIT/BSD/etc. -- I work, in my day job, for a software company doing lots of proprietary Microsoft stuff, and it's always a joy to me when I come across open source stuff; especially when there's a problem. I'm aware that people release things under MIT etc. out of the goodness of their hearts, and that's really great, however I'm also aware that businesses take this, hack on it, and never release their (potentially beneficial) code.

I understand that in a lot of cases this is a conscious decision made by developers who don't mind too much that this happens, and recognise that it means their software might be used by more people.

I'm not one of those developers. I don't like that businesses get away with this, and I see the GPL as a weapon against this antisocial behaviour. It doesn't concern me in the slightest that fewer people might use my work, or that it might cause a headache in somebody's legal department (like I said; I work in a Microsoft house, so I'm painfully aware of the aggravation it can cause to deal with licensing -- it's still worth it in my view).

I addressed it more abrasively on reddit [a while back](http://www.reddit.com/r/Python/comments/mlqnp/blessings_a_library_for_lightweight_terminal_text/c324inh).

tl;dr: I am morally aligned with the GPL, and the AGPL is a suitable vehicle for my crusading.

**Comment by ahri** [comment]
I'm not closing the issue, as I'm open to other points of view, and if you're particularly persuasive I might consider changing the license.

**Comment by sean-** [comment]
The primary reasons:

1) Most libraries with Flask are MIT licensed.

2) In fact, most recent libraries are MIT licensed. http://ostatic.com/blog/the-top-licenses-on-github

3) When developing clean room software, use of GPL/AGPL software taints the entire project. The AGPL/GPL is free, like herpes. :~]

-sc

PS I'm very staunchly anti-GPL. GPL software causes me to twitch and search for alternate software, or fork software (been there, done that - did it for Ruby-LibXML waaay back in the day).

http://libxml.rubyforge.org/svn/trunk/LICENSE

**Comment by ahri** [comment]
Okay, at this point I understand that you don't like the GPL, but I don't understand _why_ you don't like it.

Since you made some points I'll address them;

1) I don't see the relevance of most Flask libraries being MIT licensed. I already recognise that plenty of people disagree with me, and want software to be more easily taken-and-used by business. That's fine, but I specifically don't want this (see: my first reply).

2) See: Reply for (1).

3) How does my choice of license taint all of Flask? I want specifics.

You compare the GPL to herpes, so what are you actually getting at? Why is the GPL painful for you?

**Comment by sean-** [comment]
> 3) How does my choice of license taint all of Flask? I want specifics.

It taints projects that use Flask + mustache. The Flask ecosystem is MIT/BSD based, being GPL or worse, AGPL, the library excludes itself from participation in the community due to the terms of the license.

> You compare the GPL to herpes, so what are you actually getting at? Why is the GPL painful for you?

I trade my time and code for the ability to use the code that I produce however I see fit. Said better here:

http://www.itworld.com/it-managementstrategy/233753/gpl-copyleft-use-declining-faster-ever

http://blogs.the451group.com/opensource/2011/12/15/on-the-continuing-decline-of-the-gpl/

-sc

## 

Sean Chittenden
sean@chittenden.org

**Comment by ahri** [comment]
So, in summary:

The GPL and its ilk are in proportional decline in the FLOSS ecosystem, but are experiencing growth in real terms.

You're using numerical analyses, whose only speculation revolves around community-governed projects, to tell me why you, personally, feel that the GPL is painful in the same way as herpes?

You're not being very clear in your reasoning. You're not answering my question. You're ignoring the fact that I just don't care that most people are using MIT licensing or similar for their new projects: it's not an argument that persuades me (in and of itself) and repeating it only serves to make you look silly.

**Comment by sean-** [comment]
Is the GPL seeing real growth? I'd question that assertion.

GPL taints entire projects, the AGPL being the most viral of them all. I don't want to use the GPL because of it's non-free constraints.

Large portions of the python, ruby, PHP, OS, and DB community prefer non-GPL licensing. AGPL licensed software deviates from the norm. The _only_ piece of AGPL software that I use is OpenERP... why? Because it's not a building block that will ever be redistributed, it's a terminal point in a toolchain of libraries and code. All building blocks for most applications are BSD/MIT, and as such, are preferred and used more than their GPL tainted counterparts.

**Comment by ahri** [comment]
> Is the GPL seeing real growth? I'd question that assertion.

I quoted _[your source](http://blogs.the451group.com/opensource/2011/12/15/on-the-continuing-decline-of-the-gpl/)_: "this decline relates to the proportion of all open source software, while the number of projects using the GPL family has increased in real terms."

As for the rest of your post, I've addressed this point:

> I don't see the relevance of most Flask libraries being MIT licensed. I already recognise that plenty of people disagree with me, and want software to be more easily taken-and-used by business.

and 

> I just don't care that most people are using MIT licensing or similar for their new projects: it's not an argument that persuades me (in and of itself)

I'm pretty close to closing this issue, on the basis that you've had plenty of opportunity to engage in debate, and have resolutely avoided answering, or even backing up your points. You're trolling, and I'm bored of it.

**Comment by mitsuhiko** [comment]
Flask extensions will not be listed if they are GPL and will not be approved. The only exception are libraries that are depending on other GPL stuff that can't easily be replaced. This however has not happened so far.

Some older stuff I wrote about that: http://lucumr.pocoo.org/2009/2/12/are-you-sure-you-want-to-use-gpl/

**Comment by ahri** [comment]
**Status update**:

I asked the Python subreddit for its opinions on this matter, and am still considering my position.

http://reddit.com/r/Python/comments/qswpv/please_help_this_guy_argue_his_antigpl_point/

**Comment by ahri** [comment]
License changed due to feedback from reddit.

**Comment by sean-** [comment]
That's fantastic news!
